ui/app/adapters/ldap/role.js

```diff
@@ -26,12 +29,41 @@ export default class LdapRoleAdapter extends NamedPathAdapter {
     return this.getURL(backend, this.pathForRoleType(type), name);
   }
 
-  query(store, type, query) {
-    const { backend, type: roleType } = query;
-    const url = this.getURL(backend, this.pathForRoleType(roleType));
-    return this.ajax(url, 'GET', { data: { list: true } }).then((resp) => {
-      return resp.data.keys.map((name) => ({ name, backend, type: roleType }));
-    });
+  async query(store, type, query, recordArray, options) {
+    const { partialErrorInfo } = options.adapterOptions || {};
+    const { backend } = query;
+    const roles = [];
+    const errors = [];
+
+    for (const roleType of ['static', 'dynamic']) {
+      const url = this.getURL(backend, this.pathForRoleType(roleType));
```

Comment:
We could but I'm not sure what would be gained and it would make compiling the response more difficult. The only thing that is different between requests is the url so we would be copying the try/catch inside the loop into 2 different methods. I might just not be seeing it so if you have a suggestion on how to refactor let me know.

Reply:
My comment wasn't placed in a clear spot 😅 I would probably pull the stuff in the try out, so basically 
```
fetchRolesByType(roleType) {
  return this.ajax(url, 'GET', { data: { list: true } }).then((resp) => {
          return resp.data.keys.map((name) => ({ name, backend, type: roleType }));
        });
}
```
It also might look better in the actual codebase, the github line length can be tricky. But I'm ok leaving it like this too 👍